diesel_derives/src/as_changeset.rs

```diff
@@ -58,21 +40,58 @@ pub fn derive(item: syn::DeriveInput) -> Result<proc_macro2::TokenStream, Diagno
             .emit();
     }
 
-    Ok(wrap_in_dummy_mod(quote!(
-        use diesel::query_builder::AsChangeset;
-        use diesel::prelude::*;
+    let mut generate_borrowed_changeset = true;
 
-        impl #impl_generics AsChangeset for &'update #struct_name #ty_generics
-        #where_clause
-        {
-            type Target = #table_name::table;
-            type Changeset = <(#(#ref_changeset_ty,)*) as AsChangeset>::Changeset;
+    let mut direct_changeset_ty = Vec::with_capacity(fields_for_update.len());
+    let mut direct_changeset_expr = Vec::with_capacity(fields_for_update.len());
+    let mut ref_changeset_ty = Vec::with_capacity(fields_for_update.len());
+    let mut ref_changeset_expr = Vec::with_capacity(fields_for_update.len());
 
-            fn as_changeset(self) -> Self::Changeset {
-                (#(#ref_changeset_expr,)*).as_changeset()
-            }
+    for field in fields_for_update {
```

Comment:
Don't like the fact that we moved away from `map`.

Reply:
At least for me that looks ok. The corresponding block would be quite complex if put into a `.map()`